irmin.opam

```diff
@@ -26,8 +26,10 @@ depends: [
   "logs"    {>= "0.5.0"}
   "astring"
   "hex"      {with-test}
-  "alcotest" {with-test}
+  "alcotest" {>= "1.1.0" & with-test}
+  "alcotest-lwt" {with-test}
   "ppx_irmin" {with-test}
+  "irmin-mem" {with-test}
```

Comment:
will these cycle dependencies work?

Reply:
Probably not: Tracis CI say:

```
The actions to process have cyclic dependencies:
  - install irmin.dev -> install irmin-mem.dev -> install irmin.dev
```

Reply:
Works fine on OCaml-CI. IIRC, this is a limitation of the Opam solver that is lifted by 0install-solver (@talex5 may remember more than me).

I can inline the dependencies from `irmin-mem`, but it feels like admitting defeat. It seems natural for `irmin-mem` to provide implementations for use under test.

Reply:
unfortunately not admitting defeat means that this will break for every users of opam trying to install irmin :-/

Reply:
Yes :slightly_smiling_face: I've gone with a different proposed solution for now. Perhaps Opam will support this one day.

Reply:
See https://github.com/ocaml/opam/issues/4267 which tracks this issue (or at least my understanding of it :upside_down_face:).